common/logger.py

```diff
@@ -1,96 +1,133 @@
+        ts = event_dict.pop("timestamp", None)
+        if ts is not None:
+            sio.write(str(ts) + " ")
 
-            # Remove the extra 2 characters (space and comma) and add all data to the final message.
-            message += " ({})".format(params[:-2])
+        actor = event_dict.pop("actor", None)
+        if actor is not None:
+            sio.write("[" + actor + "] ")
 
-        return message
+        # force event to str for compatibility with standard library
+        event = event_dict.pop("event")
+        if not isinstance(event, str):
+            event = str(event)
 
-    @staticmethod
-    def _create_file_message(msg, **kwargs):
-        return json.dumps(_StructuredMessage(msg, **kwargs).to_dict())
+        sio.write(event)
 
-    def info(self, msg, **kwargs):
-        """
-        Logs an ``INFO`` level message to stdout and file.
+        # Represent all the key=value elements still in event_dict
+        key_value_part = " ".join(key + "=" + self._repr(event_dict[key]) for key in sorted(event_dict.keys()))
+        if len(key_value_part) > 0:
+            sio.write("\t" + key_value_part)
 
-        Args:
-             msg (:obj:`str`): the message to be logged.
-             kwargs (:obj:`dict`): a ``key:value`` collection parameters to be added to the output.
-        """
+        return sio.getvalue()
 
-        self.f_logger.info(self._create_file_message(msg, **kwargs))
-        self.c_logger.info(self._create_console_message(msg, **kwargs))
 
-    def debug(self, msg, **kwargs):
-        """
-        Logs a ``DEBUG`` level message to stdout and file.
 
-        Args:
-             msg (:obj:`str`): the message to be logged.
-             kwargs (:obj:`dict`): a ``key:value`` collection parameters to be added to the output.
-        """
+def setup_logging(log_file_path, silent=False):
+    """
+    Configures the logging options. It must be called only once, before using get_logger.
 
-        self.f_logger.debug(self._create_file_message(msg, **kwargs))
-        self.c_logger.debug(self._create_console_message(msg, **kwargs))
+    Args:
+        log_file_path(:obj:`str`): the path and name of the log file.
+        silent(:obj:`str`): if True, only critical errors will be shown to console.
 
-    def error(self, msg, **kwargs):
-        """
-        Logs an ``ERROR`` level message to stdout and file.
+    Raises:
+        (:obj:`RuntimeError`) setup_logger had already been called.
 
-        Args:
-             msg (:obj:`str`): the message to be logged.
-             kwargs (:obj:`dict`): a ``key:value`` collection parameters to be added to the output.
-        """
+    """
 
-        self.f_logger.error(self._create_file_message(msg, **kwargs))
-        self.c_logger.error(self._create_console_message(msg, **kwargs))
+    global configured
+
+    if configured:
+        raise RuntimeError("logging was already configured.")
```

Comment:
Well, I think calling this twice means that your code is broken, so it's good if the exception is not handled. Silently failing (and returning None that will likely be ignored) might lead to not detecting that you called it twice - and maybe running with a config you don't expect.

Reply:
Well, I don't think the latter would apply, since the the first time it will configure it with the original conf and the second time will not run it at all, so it would still run with the original conf.

I don't think this is a big deal though, since there is no chance this is called more than once.

If you rather leaving it like this: Cap "Logging" and remove the tailing dot

Reply:
Well, that mean we are choosing the first config, but we don't know which one was the intended one if two are given.

Updated error message in 56adb9b.